src/plugins/performance-plugin.ts

```diff
@@ -1,21 +1,25 @@
-import Perfume from 'perfume.js';
 import perfumePlugin from '@analytics/perfumejs';
 
 // Track performance metrics
-const performancePlugin = ({ sendTo }: { sendTo?: string[]}) => {
-  const destinations = Array.isArray(sendTo)
-    ? sendTo.reduce((acc, tracker) => ({ ...acc, [tracker]: true }), {})
-    : { all: true };
-  return perfumePlugin({
-    perfume: Perfume,
-    category: 'perfMetrics',
-    destinations,
-    perfumeOptions: {
-      resourceTiming: true,
-      elementTiming: true,
-      maxMeasureTime: 15000,
-    },
-  });
+const performancePlugin = ({ sendTo }: { sendTo?: string[] }) => {
+  if (typeof window !== 'undefined') {
+    import('perfume.js').then((Perfume) => {
+      const destinations = Array.isArray(sendTo)
+        ? sendTo.reduce((acc, tracker) => ({ ...acc, [tracker]: true }), {})
+        : { all: true };
+
+      return perfumePlugin({
```

Comment:
This breaks the `performancePlugin`'s contract.
Anyone using `performancePlugin` expected that it return an instance of `PerformancePlugin`. Now, it returns `void` (even if the `import` resolves).

Reply:
We should keep the `performancePlugin` returning an instance of `PerfumePlugin` but change the `performancePlugin` to return something like `Promise<PerfumePlugin>`.
Then we need to check every place that uses `performancePlugin` and make sure it "awaits" its result

Reply:
Yeah this is why I had to change the other file to filter out the `void`. The issue we've got here is that we can't use the `perfumePlugin` without `perfume` and that can't be used with SSR.

Reply:
But we still need to make the `performancePlugin` return the `perfumePlugin` result, right?